homeassistant/components/onvif/camera.py

```diff
@@ -204,8 +204,9 @@ async def _async_get_stream_uri(self) -> str:
                 self._stream_uri_future.set_exception(err)
             raise
         url = URL(uri_no_auth)
-        url = url.with_user(self.device.username)
-        url = url.with_password(self.device.password)
+        if self.device.username or self.device.password:
+            url = url.with_user(self.device.username)
+            url = url.with_password(self.device.password)
```

Comment:
This change looks good to me. Can you add a test?

Reply:
It looks like empty username and empty username is valid per https://www.ietf.org/rfc/rfc2617.txt

It seems like there are some real world cases where its a problem as well https://github.com/jaredhanson/passport-http/issues/9

Reply:
The point is, is it possible for both username and password at the same time?

Reply:
Yes